app/services/notifications/new_comment/send.rb

```diff
@@ -17,27 +17,40 @@ def call
         return if comment.score.negative?
 
         user_ids = Set.new(comment_user_ids + subscribed_user_ids + top_level_user_ids + author_subscriber_user_ids)
+        user_ids = user_ids.delete(comment.user_id)
 
         json_data = {
           user: user_data(comment.user),
           comment: comment_data(comment)
         }
 
-        targets = []
-        user_ids.delete(comment.user_id).each do |user_id|
+        user_ids.each do |user_id|
           Notification.create(
             user_id: user_id,
             notifiable_id: comment.id,
             notifiable_type: comment.class.name,
             action: nil,
             json_data: json_data,
           )
-
-          targets << "user-notifications-#{user_id}" if User.find_by(id: user_id)&.mobile_comment_notifications
         end
 
-        # Sends the push notification to Pusher Beams channels. Batch is in place to respect Pusher 100 channel limit.
-        targets.each_slice(100) { |batch| send_push_notifications(batch) }
+        targets = User.where(id: user_ids, mobile_comment_notifications: true).ids
+
+        if FeatureFlag.enabled?(:mobile_notifications)
+          # Send PNs using Rpush
+          PushNotifications::Send.call(
+            user_ids: targets,
+            title: "@#{comment.user.username}",
+            body: "re: #{comment.parent_or_root_article.title.strip}",
+            payload: { url: URL.url("/notifications/comments") },
+          )
+        else
```

Comment:
Will PusherBeams not be used if we're using mobile notifications? Don't we have to do BOTH right now until we rewrite Desktop notifications to use RPush?

Reply:
Great question because I was wondering the same!

I settled for this decision because we could start testing on other Forem instances first (not DEV). This would mean we would enable the feature flag in BHC and then gradually in the Forem cloud instances. Those are not using Pusher Beams (only DEV really is with the DEV iOS app) and I thought we could enable this in DEV the moment we get rid of Pusher Beams entirely.

I think I was doing an unnecessary early optimization by avoiding the duplicate PN delivery in case the Feature Flag in enabled. All of this to say I'm totally okay with changing this and will be pushing the correction soon 👍

Reply:
Ah, I actually wasn't clear what we were using Pusher Beams for! As long as we aren't doubling up on notifications I think it's fine to split things up and worry about removing Pusher Beams later.

Reply:
I did end up implementing the suggestion because we don't know how long it will take us to get rid of the Flag entirely and we could benefit from testing this in a higher traffic instance like DEV. 

It won't end up in duplicate PNs because Beams only sends PNs to DEV iOS and Rpush will only target Forem iOS 👍